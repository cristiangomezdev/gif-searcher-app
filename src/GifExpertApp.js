import React,{useState} from "react";

import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
import image from './Logo.png';

const GifExpertApp = ({defaultCategories=[]}) =>{

    const[categories,setCategories] = useState(defaultCategories)

return (<div>
        <h2>Gif Search Engine <img src={image} className="logo"></img><h4>Developed by Cristian Gomez</h4></h2>
        
        <h3>Search whatever you want </h3>
        <AddCategory setCategories={setCategories}/><hr/>


    <ol>{categories.map((category) => <GifGrid key={category} category={category}/> )}</ol>
    </div>);
}

export default GifExpertApp;

