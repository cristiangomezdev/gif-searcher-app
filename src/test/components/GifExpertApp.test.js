import React from 'react';
import {shallow} from 'enzyme';
import  GifExpertApp  from '../../GifExpertApp';

describe('pruebas GifExpertApp', () => {

    test('debe de mostrarse correctamente', () => {
        const  wrapper = shallow(<GifExpertApp  />);
        expect (wrapper).toMatchSnapshot();
    });
    test('Debe mostrar una lista de categorias', () => {
        const categories = ['One punch','dragonball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    });
})