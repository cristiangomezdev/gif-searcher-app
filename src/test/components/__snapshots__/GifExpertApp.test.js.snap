// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas GifExpertApp debe de mostrarse correctamente 1`] = `
<div>
  <h2>
    Gif Searcher
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
